### ISSUE TITLE
Picking the right complete basis of steerable functions

### ISSUE THREAD

**Issue by belizgunel** [ISosnovik/sesn#1]
Hi,

Really enjoyed your work, thanks for open-sourcing! I have few questions on the implementation choices that I was not able to locate in the paper:

1. How did you go about picking the complete basis of 2D Hermite polynomials with 2D Gaussian envelope? Did you try any other options, and if so, what were they? 
2. Did you use a heuristic method for picking the base parameter a for the scaling group for each dataset? I'd imagine the optimal value would be different for different datasets.

Thank you so much in advance!

Best,
Beliz

**Comment by ISosnovik** [comment]
Hi Beliz,

Thank you for the interest in our paper! It's actually a very good question. We partially answer it in our recent paper [DISCO](https://arxiv.org/pdf/2106.02733.pdf).  We demonstrate that lower equivariance error leads to better accuracy. Thus, you should choose the set of functions and their parameters, such as `a`, that minimize the equivariance error. It wil give you a good solution.

I believe that due to the stochastic nature of optimization, there can be a locally better solution though. Different hyperparameters of the basis lead to different optimization "paths". And I think it makes sense to first choose the parameters which minimize the equivariance error. It is a very fast procedure. And then just do some sort of hyperparameter tuning in the vicinity of the solution.

For SESN we were just lucky to select more-or-less good parameters. We trained a set of 5 (if I remember it correctly) networks with different hyperparameters and chose the best one.
